pageserver/src/http/routes.rs

```diff
@@ -1757,27 +1759,36 @@ async fn timeline_detach_ancestor_handler(
 
         let timeline = tenant.get_timeline(timeline_id, true)?;
 
-        let (_guard, prepared) = timeline
+        let progress = timeline
             .prepare_to_detach_from_ancestor(&tenant, options, ctx)
             .await?;
 
-        let res = state
-            .tenant_manager
-            .complete_detaching_timeline_ancestor(tenant_shard_id, timeline_id, prepared, ctx)
-            .await;
-
-        match res {
-            Ok(reparented_timelines) => {
-                let resp = pageserver_api::models::detach_ancestor::AncestorDetached {
+        // uncomment to allow early as possible Tenant::drop
+        // drop(tenant);
```

Comment:
Any reason why you've not uncommented this? It makes sense to do so since we do IO further on

Reply:
Yes, definitely it'd be better, it's related to the #8229 work though, maybe I'll include it in one of those PRs.

Reply:
Keeping it commented out so I will not forget this :)